src/migration/1721310640980-PostStatsTriggers.ts

```diff
@@ -0,0 +1,69 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { VoteTriggers1692265909018 } from "./1692265909018-VoteTriggers";
+
+export class PostStatsTriggers1721310640980 implements MigrationInterface {
+    name = 'PostStatsTriggers1721310640980'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`
+          CREATE OR REPLACE FUNCTION user_post_vote_insert_trigger_function()
+          RETURNS TRIGGER AS $$
+          BEGIN
+              IF NEW.vote = 1 THEN
+                  UPDATE post SET upvotes = upvotes + 1, "statsUpdatedAt" = NOW() WHERE id = NEW."postId";
```

Comment:
Same trigger functions only adding `statsUpdatedAt` update.

Reply:
> I'd imagine the statsUpdatedAt as an independent trigger that checks if upvotes, comments, or downvotes have changed, if so it updated the timestamp.
Just to keep every trigger very focused. Wdyt?

@idoshamun commenting here so we can talk in a thread.

I thought about it but it would introduce another UPDATE call to `post` table which would again trigger CDC so opted for this approach.

Reply:
Are you sure? I think we can manipulate the payload within the trigger so we can get away with a single statement without introducing an overhead. With a before update trigger i'm pretty certain it's possible

Reply:
You mean `NEW` and `OLD` inside triggers?

Reply:
Yes the `NEW` specifically on the before update trigger. I think we have some examples around here

Reply:
But in this one `NEW` is a comment entity, since trigger is on comment table, and we need to update it on post entity

Reply:
Triggers would have to be on post table so I can do `NEW.statsUpdatedAt = NOW()`?

Reply:
Ignore my comments, yeah you are right, we are doing it for `votedAt`, I'll do it like that.